main/src/ca/uwaterloo/flix/util/Grammar.scala

```diff
@@ -26,4 +26,38 @@ object Grammar {
       case (end, suffix) if n.toString.endsWith(end) => n.toString + suffix
     }.getOrElse(n.toString + defaultSuffix)
   }
+
+  /**
+    * Returns a string representation of a count of `n` things called `name` that optionally `are` something.
+    * When `n` is 1, the sentence will be singular, otherwise it will be plural.
+    *
+    * It is assumed that `name` is a regular noun, that can be pluralized by appending an "s".
+    *
+    * For example, given `count(1, Some("cat"), are=true)` returns "1 cat is",
+    * and given `count(2, Some("cat"), are=true)` returns "2 cats are".
+    *
+    * The `name` argument is optional, for example `count(2, None, are=true)` returns "2 are".
+    *
+    * The `are` parameter can be false (default), for example `count(3, Some("cat"))` returns "3 cats".
+    */
+  def count(n: Int, name: Option[String], are: Boolean = false): String = {
```

Comment:
I propose we separate this into two functions, one for "is"/"are" and one for noun pluralization.

Reply:
Sounds good.